### PR TITLE
LUN-427: Optimize for tablet

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

Please check all screen properly on android!! There is one small design difference for BottomTabs
Seems like the lib auto checks the space and put label and icons next to each other.
I think it looks fine for tablet? If not i can even change that

image

### CODE REVIEW

File: release-notes/unreleased/LUN-427-optimize-for-tablet.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+issue_key: LUN-427
+show_in_stores: true
+platforms:
+  - android
+  - ios
+de: Lunes wurde für Tablets optimiert
```
Comment (SteffiStuffel) [inline release-notes/unreleased/LUN-427-optimize-for-tablet.yml:6]: Maybe `Optimierung für Tablets`. It's somehow seems strange to me to have the app name in a release note.

**Review by SteffiStuffel — COMMENTED** [review]
Why is `hp` better than `wp`? A tablet is both heigher and wider.

Few small things i saw:
Padding of article hint is to big.
LUN-427-articlehint

Text-boxes border looks weird in feedback and search
LUN-427-feedback
LUN-427-leikon

Menu is way to small
LUN-427menu

No results icon and text is too small
LUN-427-no-results

**Comment by f1sh1918** [comment]
> Why is `hp` better than `wp`? A tablet is both heigher and wider.
> 
> Few small things i saw: Padding of article hint is to big. LUN-427-articlehint
> 
> Text-boxes border looks weird in feedback and search LUN-427-feedback LUN-427-leikon
> 
> Menu is way to small LUN-427menu
> 
> No results icon and text is too small LUN-427-no-results

Well thats what i explained in my presentation. If you use wp everything will scale 200%. Just put a mobile phone on a tablet (both portrait) and it will be self explaining.
Well strange border behavior on android...doesn't happen on ios :/ gonna check that

**Comment by SteffiStuffel** [comment]
> 
> Well thats what i explained in my presentation. If you use wp everything will scale 200%. Just put a mobile phone on a tablet (both portrait) and it will be self explaining. Well strange border behavior on android...doesn't happen on ios :/ gonna check that

What do you men bei "put a mobile phone on a tablet"?

**Comment by f1sh1918** [comment]
> > Well thats what i explained in my presentation. If you use wp everything will scale 200%. Just put a mobile phone on a tablet (both portrait) and it will be self explaining. Well strange border behavior on android...doesn't happen on ios :/ gonna check that
> 
> What do you mean bei "put a mobile phone on a tablet"?

image

Now you can image the scaling factor for width & height(i resized my tablet)

**Comment by SteffiStuffel** [comment]
> > > Well thats what i explained in my presentation. If you use wp everything will scale 200%. Just put a mobile phone on a tablet (both portrait) and it will be self explaining. Well strange border behavior on android...doesn't happen on ios :/ gonna check that
> > 
> > 
> > What do you mean bei "put a mobile phone on a tablet"?
> 
> 
> Now you can image the scaling factor for width & height(i resized my tablet)

But as width differs more, woundn't width make more sense to use it for scaling or what am I missing?

**Comment by f1sh1918** [comment]
> > > > Well thats what i explained in my presentation. If you use wp everything will scale 200%. Just put a mobile phone on a tablet (both portrait) and it will be self explaining. Well strange border behavior on android...doesn't happen on ios :/ gonna check that
> > > 
> > > 
> > > What do you mean bei "put a mobile phone on a tablet"?
> > 
> > 
> > Now you can image the scaling factor for width & height(i resized my tablet)
> 
> But as width differs more, woundn't width make more sense to use it for scaling or what am I missing?

the point is you have to reflect the distance between the screen and you use a tablet in the same distance as a smart phone (not like laptop or monitor) if you increase everything 2x its way too big and looks ugly. You can try by yourself and change everything to wp and revert to the old percentage values...

**Review by SteffiStuffel — APPROVED** [review]
Looks good, have not tested again, as my emulator is not working.

**Review by SteffiStuffel — CHANGES_REQUESTED** [review]
I tested it on Pixel 4:

1) Images are no longer full width. left is the current main and right is LUN-427
lun-427-image-width

2) Header has an margin, can also be seen in 1)
lun-427-header height

3) Icon of modal is too big (Happens in all modals e.g. cancel, feedback..)
lun-427-modal

4) When pressing on "cancel-exercise" icon in the header, it moved, while being pressed:
lun-427-on-press

**Comment by f1sh1918** [comment]
> I tested it on Pixel 4:
> 
> 1. Images are no longer full width. left is the current main and right is LUN-427
>   
> 2. Header has an margin, can also be seen in 1)
>  
> 3. Icon of modal is too big (Happens in all modals e.g. cancel, feedback..)
>   
> 4. When pressing on "cancel-exercise" icon in the header, it moved, while being pressed:
>   
@SteffiStuffel 
1. this relies to this lib `react-native-image-zoom-viewer` and i have no clue how to fix for android (for ios everything is fine). Does not rely on my changes. Since there was no update since 2 years, we should think about moving to another lib tbh.
2. can not reproduce with my Pixel 4 Emulator nor real device
3.  fixed icon size
4. can not reproduce on emulator nor real device

image

@steffenkleinle can you please test on emulator & real device, since steffi and me got different results on emulators which is very strange tbh

**Review by sarahsporck — COMMENTED** [review]
I think instead of taking half the wp value consider using about 0.6 of the wp. This seems to keep the proportions the same on at least my device. If it's a different factor for you we may have to re-discuss this. 

I'd also request that the images should use the full width available.

I think when adjusting to 0.6 of the wp value most issues will disappear from alone. 

I did not really look

File: src/components/FavoriteButton.tsx [hunk 2]
```diff
@@ -11,12 +11,12 @@ import { reportError } from '../services/sentry'
...
-  min-width: ${wp('9%')}px;
-  min-height: ${wp('9%')}px;
+  min-width: ${hp('4%')}px;
+  min-height: ${hp('4%')}px;
 `
 const IconOutline = styled(StarCircleIconGrey)`
-  min-width: ${wp('9%')}px;
-  min-height: ${wp('9%')}px;
+  min-width: ${hp('4%')}px;
+  min-height: ${hp('4%')}px;
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/components/FavoriteButton.tsx]: well the correct factor is `0.56` if you compare width to height for iPhone5SE (320x560)
I don't think that this gonna fix the mentioned issues ;)

[the remark this answers is not in the document]
Comment (f1sh1918) [reply src/components/FavoriteButton.tsx]: fixed this by using our spacings vars instead of hp `spacings.ts`

**Comment by f1sh1918** [comment]
> 4. When pressing on "cancel-exercise" icon in the header, it moved, while being pressed:
>  ok i found the issue tested on pixel 4, please recheck

**Comment by f1sh1918** [comment]
There is some weird behavior with the favorite button and the audio player. i have to check that maybe it relies on the svgs itself

**Review by sarahsporck — COMMENTED** [review]
I still feel as if the buttons are a little bit flat. However it is better now. What do you think @SteffiStuffel?

Further, is there a fixed process on how to calculate the hp values? E.g. if the design suggest a width of `100px` how would i calculate the corresponding hp value. Do we have this process documented somewhere?

**Comment by f1sh1918** [comment]
> I still feel as if the buttons are a little bit flat. However it is better now. What do you think @SteffiStuffel?
> 
> Further, is there a fixed process on how to calculate the hp values? E.g. if the design suggest a width of `100px` how would i calculate the corresponding hp value. Do we have this process documented somewhere?

We should define the percent of 1px i would add a comment to `spacings.ts`

**Review by sarahsporck — APPROVED** [review]
Tested on Android emulator. Really like the consistency in both tablet and mobile devices. Great Work!

**Pull request merged**